tests/unit/autocomplete/test_autogen.py

```diff
@@ -556,3 +559,61 @@ def test_remove_operations_with_required_params(self):
         # The operations dict should be empty because the FooBarThing has
         # a required parameter and we don't support that yet.
         self.assertEqual(completion_data['operations'], {})
+
+    def test_can_reference_multiple_identifiers_if_used(self):
+        custom_model = {
+            'metadata': {},
+            'operations': {
+                'ListFooBarThings': {
+                    'output': {'shape': 'ListFooBarThingsResponse'},
+                },
+                'DeleteFooBarThing': {
+                    'input': {'shape': 'DeleteFooBarThingRequest'},
+                }
+            },
+            'shapes': {
+                'DeleteFooBarThingRequest': {
+                    'type': 'structure',
+                    'members': {
+                        'FooBarThingId': {'shape': 'String'},
+                        'FooBarThingArn': {'shape': 'String'},
+                    }
+                },
+                'ListFooBarThingsResponse': {
+                    'type': 'structure',
+                    'members': {
+                        'FooBarThings': {'shape': 'FooBarThingList'},
+                    }
+                },
+                'FooBarThingList': {
+                    'type': 'list',
+                    'member': {'shape': 'FooBarThingType'},
+                },
+                'FooBarThingType': {
+                    'type': 'structure',
+                    'members': {
+                        # The DeleteFooBarThing accepts either an Id
+                        # or an Arn.
+                        'FooBarThingId': {'shape': 'String'},
+                        'FooBarThingArn': {'shape': 'String'},
+                        # However it doesn't accept a "Name", so this
+                        # identifier will be pruned from the response.
+                        'FooBarThingName': {'shape': 'String'},
```

Comment:
It might be better to make the name of the member more explict in that it will be pruned (i.e. `FoorBarThingPrunedId`)